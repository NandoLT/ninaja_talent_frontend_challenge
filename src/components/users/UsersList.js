import React, { useState } from 'react';
import UserDetail from './UserDetail';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faEye } from '@fortawesome/free-solid-svg-icons';

import '../../assets/css/usersList.css';

const UsersList= ({ users }) => {

    const [seeUserDetail, setSeeUserDetail] = useState(false);
    const [userDetail, setUserDetail] = useState({});

    const handleUserDetail = () => {
        setSeeUserDetail(!seeUserDetail);
    }

    const getUserDetail = (user) => {
        setUserDetail(user);
    }

    return (
        <>
            <div className="users-table">
                <table className="table is-striped is-hoverable">
                    <thead>
                        <tr>
                            <th> User Id</th>
                            <th> User Name </th>
                            <th></th>
                        </tr>
                    </thead>
                    <tbody>
                        {users.map( user => { 
                            return (
                                <tr key={ user.id }>
                                    <td>{ user.id }</td>
                                    <td>{ user.name }</td>
                                    <td><FontAwesomeIcon 
                                            className="see_user_detail"
                                            icon={ faEye }
                                            onClick={ () => { 
                                                handleUserDetail();
                                                getUserDetail(user);
                                            }}
                                        />
                                    </td>
                                </tr>
                            )
                        })}
                    </tbody>
                </table>
            </div>
            <UserDetail seeUserDetail={ seeUserDetail } handleUserDetail={ handleUserDetail } userDetail={ userDetail }/>
        </>
    );
}


export default UsersList;

