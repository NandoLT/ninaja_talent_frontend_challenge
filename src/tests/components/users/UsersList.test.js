import { shallow } from 'enzyme';
import React from 'react';
import UsersList from '../../../components/users/UsersList';

describe('Tests in <UsersList />', () => {
    test('Should display correctly list of users if users array is not empty', () => {
        const users = [
            {
                'id': 23,
                'name': 'Fernando',
                'mail': 'fernadno@ninjatalent.es'
            }
        ]
        const wrapper = shallow(<UsersList users={ users } />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Should display <EmptyLisst /> if users is empty', () => {

        const users = [];
        const wrapper = shallow(<UsersList users={ users } />);
        expect(wrapper.find('EmptyList').length).toBe(1);
    })
})