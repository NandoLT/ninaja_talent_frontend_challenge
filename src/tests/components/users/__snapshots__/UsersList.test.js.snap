// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <UsersList /> Should display correctly list of users if users array is not empty 1`] = `
<Fragment>
  <div
    className="users-table"
  >
    <table
      className="table is-striped is-hoverable"
    >
      <thead>
        <tr>
          <th>
             User Id
          </th>
          <th>
             User Name 
          </th>
          <th />
        </tr>
      </thead>
      <tbody>
        <tr
          key="23"
        >
          <td>
            23
          </td>
          <td>
            Fernando
          </td>
          <td>
            <FontAwesomeIcon
              border={false}
              className="see_user_detail"
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    576,
                    512,
                    Array [],
                    "f06e",
                    "M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z",
                  ],
                  "iconName": "eye",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              onClick={[Function]}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <UserDetail
    handleUserDetail={[Function]}
    seeUserDetail={false}
    userDetail={Object {}}
  />
</Fragment>
`;
